kernel/libs/aster-bigtcp/src/socket/bound.rs

```diff
@@ -752,6 +772,16 @@ impl<E: Ext> TcpConnectionBg<E> {
     pub(crate) fn is_dead(&self) -> bool {
         self.inner.is_dead()
     }
+
+    pub(crate) const fn connection_key(&self) -> ConnectionKey {
```

Comment:
`ConnectionKey`/`ListenerKey` is a `Copy`, but it's also a bit bigger (and when we support IPv6 it will be even bigger).

So how about making `connection_key()`/`listener_key()` return `&ConnectionKey`/`&ListenerKey`? The caller may want to copy it, but it's at the caller's convenience.

This also means that we don't have to take the reference with us when we pass it to other methods.
```rust
backlog.connecting.remove(&this.connection_key())
```